code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/InternalMessagingUtils.java

```diff
@@ -560,6 +562,83 @@ static String getSharedStateEcid(final Map<String, Object> edgeIdentityState) {
+     * @param newPushToken A {@code String} containing the push token to be synced
+     * @return {@code boolean} indicating if the push token should be synced
+     */
+    static boolean shouldSyncPushToken(
+            @Nullable final Map<String, Object> configSharedState,
+            @Nullable final String newPushToken) {
+        if (StringUtils.isNullOrEmpty(newPushToken)) {
+            Log.debug(
+                    MessagingConstants.LOG_TAG,
+                    "shouldSyncPushToken",
+                    "New push token is null or empty, push token will not be synced.");
+            return false;
+        }
+
+        final boolean isPushRegistrationPaused =
+                DataReader.optBoolean(
+                        configSharedState,
+                        MessagingConstants.SharedState.Configuration.PUSH_REGISTRATION_PAUSED,
+                        false);
+
+        if (isPushRegistrationPaused) {
+            Log.debug(
+                    MessagingConstants.LOG_TAG,
+                    "shouldSyncPushToken",
+                    "Push registration is paused, push token will not be synced.");
+            return false;
+        }
+
+        final String existingPushToken = getExistingPushToken();
+        if (existingPushToken != null && existingPushToken.equals(newPushToken)) {
+            Log.debug(
+                    MessagingConstants.LOG_TAG,
+                    "shouldSyncPushToken",
+                    "Existing push token matches the new push token, push token will not be"
+                            + " synced.");
+            return false;
+        }
+
+        Log.debug(
+                MessagingConstants.LOG_TAG,
+                "shouldSyncPushToken",
+                "Existing push token not found or the push token is new. The push token will be"
+                        + " synced.");
+
+        // persist the push token in the messaging named collection
+        final NamedCollection messagingNamedCollection = getNamedCollection();
+        messagingNamedCollection.setString(
+                MessagingConstants.SharedState.Messaging.PUSH_IDENTIFIER, newPushToken);
+
+        return true;
+    }
+
+    // ========================================================================================
+    // Datastore utils
+    // ========================================================================================
+    /**
+     * Retrieves the Messaging extension {@link NamedCollection} from the {@link DataStoring}
+     * service.
+     *
+     * @return The Messaging extension {@link NamedCollection}.
+     */
+    private static NamedCollection getNamedCollection() {
+        final DataStoring dataStoreService = ServiceProvider.getInstance().getDataStoreService();
+        return dataStoreService.getNamedCollection(MessagingConstants.DATA_STORE_NAME);
+    }
+
+    @Nullable static String getExistingPushToken() {
```

Comment:
Rename to `getPushTokenFromPersistence` to make it clear this is a stored value.